docs/proposals/scheduler/controller.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+
+// ------------------
+// Scheduling
+// ------------------
+
+message SchedulingRequest {
+    // definition of the workload
+}
+
+enum SchedulingStatus {
+    SCHEDULED = 0;
+    // the request was malformed or safety checks failed
+    BAD_REQUEST = 1;
+    // no underlying agent could be contacted to schedule
+    NO_CONTACTABLE_AGENT = 2;
+    // all contacted agents refused the workload
+    AGENTS_REFUSED_OPERATION = 3;
+}
```

Comment:
Here you're mixing status and reasons. Status for a workload scheduling request could be:

```proto
enum SchedulingStatus {
   SCHEDULED = 0;
   SCHEDULING = 1;
   REJECTED = 2;
}
```

And then you could define a set of scheduling error/rejection statuses and carry one in the `SchedulingResponse`.

Reply:
[Done](https://github.com/dev-sys-do/orka/pull/2/commits/dae37730c6d56acaa1cfbe39fe77b81d9714271b#diff-de3632c29b02edb1da9170fa6071371537f5980fb145bb4c842e69e2094a41c9R31-R46)! Status code and rejection reasons have been separated in two distinct fields.